### ISSUE TITLE
A possible bug in VQH training

### ISSUE THREAD

**Issue by Smadx** [hume-vla/hume#14]
In [src](https://github.com/hume-vla/hume/tree/main/src)/[hume](https://github.com/hume-vla/hume/tree/main/src/hume)/[training](https://github.com/hume-vla/hume/tree/main/src/hume/training)/dataset.py line 144

``prev_ret`` has not been updated
```python
def construct_reward_data(self):
        reward_keys = ["reward", "mc", "tdm", "done"]
        reward_data = {k: [] for k in reward_keys}

        from datasets import Dataset
        from tqdm import tqdm

        for i in tqdm(range(self.num_episodes)):
            ep_start = self.episode_data_index["from"][i].item()
            ep_end = self.episode_data_index["to"][i].item()

            reward = torch.zeros(ep_end - ep_start, dtype=torch.float32)
            reward[: -self.num_pos] = -1.0
            td_mask = reward.bool()

            mc_ret = torch.zeros_like(reward)
            prev_ret = 0.0
            for t in range(len(reward) - 1, -1, -1):
                mc_ret[t] = reward[t] + self.discount * prev_ret * td_mask[t] # line 144

            done = torch.zeros(ep_end - ep_start, dtype=torch.bool)
            done[-1] = True

            for k, v in zip(reward_keys, [reward, mc_ret, td_mask, done]):
                reward_data[k].append(v)
```

Besides, in [src](https://github.com/hume-vla/hume/tree/main/src)/[hume](https://github.com/hume-vla/hume/tree/main/src/hume)/[models](https://github.com/hume-vla/hume/tree/main/src/hume/models)/modeling_hume.py line 581

the last param for ``value_query_head.forward`` should be "masks", but ``batch["reward.vqh"]`` was passed:

```python
 temperature_loss, policy_loss, critic_loss, log_dict = (
            self.value_query_head.forward(
                images,
                img_masks,
                lang_tokens,
                lang_masks,
                vqh_images,
                vqh_img_masks,
                batch["action"][:, : self.config.vqh_chunk_size, :],
                batch["reward.vqh"],
                batch["mc.vqh"],
                batch["reward.vqh"].to(dtype=torch.float), # line 581
            )
        )
```
I wonder if this will affect the effect of vqh. Maybe they should be fixed
